### ISSUE TITLE
Last line of email/diff always red since 7.8

### ISSUE THREAD

**Issue by petrcech** [gitlabhq/gitlabhq#8976]
Hi,
since upgrade of gitlab from version 7.7 to 7.8 the notification emails have the last line of diff marked red, e.g.
```
<pre><pre class="highlight"><code><span class="gd">--- a/XXX
</span><span class="gi">&#43;&#43;&#43; b/XXX
</span><span class="gh">@@ -1,7 &#43;1,7 @@
</span> 
 !
<span class="gd">-! Last configuration change at 20:49:49 CET Wed Mar 18 2015 by xxx
-! NVRAM config last updated at 20:51:46 CET Wed Mar 18 2015 by xxx
</span><span class="gi">&#43;! Last configuration change at 20:54:51 CET Wed Mar 18 2015
&#43;! NVRAM config last updated at 21:12:36 CET Wed Mar 18 2015 by xxx
</span> !
 version 15.2
<span class="err"> no service pad</span></code></pre></pre>
<br>
```

GUI ist unaffected. The issue also exists in the latest 7.8.4 release.

Regards
Petr

**Comment by rumpelsepp** [comment]
I think it should be reported here: https://github.com/rumpelsepp/rugments

Please open an issue there and I'll have a look at it.

**Comment by DouweM** [comment]
The problem still exists, but I don't think it's Rugments's fault after all. For some reason the final newline of the diff is stripped by GitLab, which causes Rugments to render it as an error, which should only happen for `\No new line at end of file` (which we expect as an error). GitLab shouldn't touch the diff's final newline, I will look into it.

cc @rumpelsepp

**Comment by DouweM** [comment]
Fixed in https://gitlab.com/gitlab-org/gitlab-ce/merge_requests/480, which pulls in gitlab_git 7.1.5 which includes https://gitlab.com/gitlab-org/gitlab_git/merge_requests/18, which fixes the issue with diffs.

**Comment by rumpelsepp** [comment]
:+1: So this one is finally fixed now?

**Comment by DouweM** [comment]
@rumpelsepp It will be when that MR is merged :)

**Comment by DouweM** [comment]
Fixed in 18dfe0ce76121a9a6f5a83588590b05acc9ff7b8.
